src/query/ast/src/parser/expr.rs

```diff
@@ -1545,71 +1545,129 @@ pub fn expr_element(i: Input) -> IResult<WithSpan<ExprElement>> {
-            rule!(
-                #substring : "`SUBSTRING(... [FROM ...] [FOR ...])`"
-                | #trim_from : "`TRIM([(BOTH | LEADEING | TRAILING) ... FROM ...)`"
-                | #is_distinct_from: "`... IS [NOT] DISTINCT FROM ...`"
-                | #chain_function_call : "x.function(...)"
-                | #list_comprehensions: "[expr for x in ... [if ...]]"
-                | #count_all_with_window : "`COUNT(*) OVER ...`"
-                | #function_call
-                | #escape: "`ESCAPE '<escape>'`"
-            ),
-            rule!(
-                #case : "`CASE ... END`"
-                | #tuple : "`(<expr> [, ...])`"
-                | #subquery : "`(SELECT ...)`"
-                | #stage_location: "@<location>"
-                | #column_ref : "<column>"
-                | #dot_access : "<dot_access>"
-                | #map_access : "[<key>] | .<key> | :<key>"
-                | #literal : "<literal>"
-                | #current_date: "CURRENT_DATE"
-                | #current_time: "CURRENT_TIME"
-                | #current_timestamp: "CURRENT_TIMESTAMP"
-                | #array : "`[<expr>, ...]`"
-                | #map_expr : "`{ <literal> : <expr>, ... }`"
-            ),
-        ))),
-        |(span, elem)| WithSpan { span, elem },
-    )(i)
+    try_dispatch!(i, true,
+        IS => with_span!(rule!(#is_null | #is_distinct_from)).parse(i),
+        NOT => with_span!(rule!(
+            #in_list
+                | #in_subquery
+                | #exists
+                | #between
+                | #binary_op
+                | #unary_op
+        ))
+        .parse(i),
+        IN => with_span!(rule!(#in_list | #in_subquery)).parse(i),
+        LIKE => with_span!(rule!(#like_subquery | #binary_op)).parse(i),
+        EXISTS => with_span!(exists).parse(i),
+        BETWEEN => with_span!(between).parse(i),
+        CAST | TRY_CAST => with_span!(cast).parse(i),
+        DoubleColon => with_span!(pg_cast).parse(i),
+        POSITION => with_span!(position).parse(i),
+        IDENTIFIER => {
+            return with_span!(column_ref).parse(i);
+        },
+        IdentVariable => with_span!(variable_access).parse(i),
+        ESCAPE => with_span!(escape).parse(i),
+        COUNT => with_span!(rule!{ #count_all_with_window | #function_call}).parse(i),
+        SUBSTRING | SUBSTR => with_span!(substring).parse(i),
+        TRIM => with_span!(trim_from).parse(i),
+        CASE => with_span!(case).parse(i),
+        LParen => with_span!(rule!(#tuple | #subquery)).parse(i),
+        ANY | SOME | ALL => with_span!(subquery).parse(i),
+        Dot => {
+            return with_span!(rule!(#chain_function_call | #dot_access | #dot_number_map_access))
+                .parse(i);
+        },
+        Colon => {
+            return with_span!(colon_map_access).parse(i);
+        },
+        LBracket => {
+            return with_span!(rule!(
+                #list_comprehensions | #bracket_map_access | #array
```

Comment:
We can combine #bracket_map_access and #array into one and perform the transform in the postfix function